hazelcast/src/main/java/com/hazelcast/internal/services/WanSupportingService.java

```diff
@@ -38,14 +39,15 @@ public interface WanSupportingService {
     void onReplicationEvent(InternalWanEvent event, WanAcknowledgeType acknowledgeType);
 
     /**
-     * Processes a WAN sync batch.
+     * Processes a WAN sync batch asynchronously.
      *
      * @param batch           collection of events, which represents a batch
      * @param acknowledgeType determines should this method wait for the event to be processed fully
      *                        or should it return after the event has been dispatched to the
      *                        appropriate member
+     * @return the future to indicate processing of all entries in the batch
      */
-    void onSyncBatch(Collection<InternalWanEvent> batch, WanAcknowledgeType acknowledgeType);
+    CompletableFuture<Void> onSyncBatch(Collection<InternalWanEvent> batch, WanAcknowledgeType acknowledgeType);
```

Comment:
`CompletionStage<Void>` seems sufficient as return type here; no reason for this API's client code to be able to complete the future themselves.

Reply:
Do you mean redundant? Like: 

> CompletionStage<Void> seems ~sufficient~ redundant as return type here; no reason for this API's client code to be able to complete the future themselves.

If not, can you elaborate more?

Reply:
https://github.com/hazelcast/hazelcast/pull/23919/commits/a5f2a63a06e401addb928ecb2b288b55f7748c37